README.md

```diff
@@ -185,18 +185,18 @@ fn await_foo(py: Python, foo: PyObject) -> PyResult<()> {
 
             // if it errors we know we're done.
             Err(e) => {
-                // We have to strip off the exception name because its added by PyO3
-                // making it like "StopIteration: some value"
-                let unclean_value = e.pvalue(py).to_string();
-
-                // There is probably a much better method of doing this.
-                let clean_value = unclean_value.trim_start_matches("StopIteration: ");
-
-                // Lets display that result of ours
-                println!("Result of our coroutine: {:?}", clean_value);
-
-                // Time to escape from this while loop.
-                return Ok(());
+                if let Ok(stop_iteration) = e.pvalue(py).extract::<PyStopIteration>() {
```

Comment:
Rereading this it should have been `.downcast` because we're targeting a Python-native type.

```suggestion
                if let Ok(stop_iteration) = e.pvalue(py).downcast::<PyStopIteration>() {
```